### ISSUE TITLE
Some first-time user problems

### ISSUE THREAD

**Issue by luizirber** [qri-io/qri#512]
Hi! I learned about `qri` at the dweb summit during the data stewardship talk, and hit some problems while trying to set it up.

## Building

I'm running Arch Linux, so I tried to follow the [install instructions]() in this repo. They... don't work (lots of `gx` errors). Maybe it's better to point to "clone this repo, run `make`" instead?

## Setup
Since the CLI tutorial is ["coming soon"](http://qri.io/docs/tutorials/cli_quickstart/), I checked the help message and tried to run `qri setup`. I was surprised when it tried to use my IPFS default repo. I'm aware that this is probably not the initial experience for someone who never learned about `IPFS` before, but maybe move the `qri` IPFS repo into `$HOME/.qri`? (I ended up running `IPFS_PATH=~/.qri/repo qri setup`)

(maybe this was supposed to be seamless, but I'm running a repo which is already on `7` and `qri` is in an IPFS version that expects `6`).

```bash
$ ./qri setup  
choose a peername: [orange_red_shih_tzu]: 

error creating IPFS repo: ipfs daemon is running. please stop it to run this command

$ systemctl --user stop ipfs

$ ./qri setup               
choose a peername: [sapphire_blue_airedale_terrier]: 

repo not available (are you running qri in another terminal?)

$ ./qri setup  
repo already initialized

$ ./qri setup --remove
error opening local filestore ipfs repository: Your programs version (6) is lower than your repos (7).
Please update ipfs to a version that supports the existing repo, or run
a migration in reverse.

See https://github.com/ipfs/fs-repo-migrations/blob/master/run.md for details.

$  rm -rf ~/.qri 

$ IPFS_PATH=~/.qri/repo ./qri setup
choose a peername: [razzle_dazzle_rose_american_eskimo_dog]: 

set up qri repo at: /home/luizirber/.qri
```

**Comment by b5** [comment]
👋, thanks so much for this detailed feedback @luizirber! A bunch of the team is on holidays for this week & next, so some of the work mentioned here may not be accomplished for a while.

> I'm running Arch Linux, so I tried to follow the install instructions in this repo. They... don't work (lots of gx errors). Maybe it's better to point to "clone this repo, run make" instead?

We've got a long way to go on our linux install story (not nearly as long as windows 😉). Totally agreed on the "clone this repo, run make" change. In the future I'd like to include tarballs that have all deps with each release. Looks like we have two 

> Since the CLI tutorial is "coming soon" ...

we'll prioritize a small CLI tutorial next for documentation work.

> but maybe move the qri IPFS repo into $HOME/.qri? (I ended up running IPFS_PATH=~/.qri/repo qri setup)

Thanks so much for this feedback. I go back & forth on using the default IPFS repo or not, and gathering user expectations really helps here. First we'll be upgrading our code to the lastest go-ipfs release in the second half of August, which will resolve the repo issue. As for using a completely separate repo for qri's IPFS stuff, it's a discussion we should absolutely have, I think it may solve many problems, but will create others, so it's worth weighing the pros & cons.

With that, I think this is the minimum feedback to incorporate to call this issue closed:
* [ ] Open an issue to discuss default IPFS repo / IPFS_PATH usage
* [ ] Open an issue for attaching tarballs with deps to our releases
* [ ] update docs for linux installation instructions that tell users to "clone & make"
* [ ] Get a bare-bones CLI tutorial up & running
* [ ] Update our deps to the latest go-ipfs (min version 0.4.17)

Thanks again Luiz, your feedback is truly appreciated!

**Comment by b5** [comment]
This issue has been around for a while, here's a report back before we close it:
* I've opened #644 to address attaching tarballs to releases, we have a clear path to getting that started, just need to do the work
* we've included instructions for [building from source](https://github.com/qri-io/qri#building-from-source) that should generally help with installation.
* re: default IPFS repo usage, this is a more complicated question I'd like to take on as part of a broader transition to being able to have Qri work out-of-process with IPFS and instead use the HTTP core-api, which is under active development. This one is going to take a while, but for the foreseeable future, I'd like to honor the `IPFS_PATH` value, and only work with that. I don't think Qri should be creating a separate IPFS repo

Thanks again for taking the time to file an issue @luizirber. It's taken us a while but we have managed to incorporate your feedback. Much appreciated.
